GUARD AH2 OT-insert wild puVar5 (puStack_14, 8sp x4)

(rationale: see preceding patch in the series)

--- a/dd2.c
+++ b/dd2.c
@@ -7232,9 +7232,11 @@
         puVar5 = (undefined4 *)
                  (((int)((__otz + (__otz >> 0x1f) * -4) - (uint)((__otz >> 0x1f) << 1 < 0)) >> 2) *
                   4 + iVar4);
+        if ((uintptr_t)puVar5 >= 0x400000u && (uintptr_t)puVar5 < 0x900000u) {
         uVar2 = *puVar5;
         *puVar5 = puStack_14;
         *puStack_14 = uVar2;
+        }
       }
     }
     puStack_14 = puStack_14 + 6;
@@ -7389,9 +7391,11 @@
         puVar5 = (undefined4 *)
                  (((int)((__otz + (__otz >> 0x1f) * -4) - (uint)((__otz >> 0x1f) << 1 < 0)) >> 2) *
                   4 + iVar4);
+        if ((uintptr_t)puVar5 >= 0x400000u && (uintptr_t)puVar5 < 0x900000u) {
         uVar2 = *puVar5;
         *puVar5 = puStack_14;
         *puStack_14 = uVar2;
+        }
       }
     }
     puStack_14 = puStack_14 + 6;
@@ -9357,9 +9361,11 @@
         puVar5 = (undefined4 *)
                  (((int)((__otz + (__otz >> 0x1f) * -4) - (uint)((__otz >> 0x1f) << 1 < 0)) >> 2) *
                   4 + iVar4);
+        if ((uintptr_t)puVar5 >= 0x400000u && (uintptr_t)puVar5 < 0x900000u) {
         uVar2 = *puVar5;
         *puVar5 = puStack_14;
         *puStack_14 = uVar2;
+        }
       }
     }
     puStack_14 = puStack_14 + 9;
@@ -10209,9 +10215,11 @@
         puVar5 = (undefined4 *)
                  (((int)((__otz + (__otz >> 0x1f) * -4) - (uint)((__otz >> 0x1f) << 1 < 0)) >> 2) *
                   4 + iVar4);
+        if ((uintptr_t)puVar5 >= 0x400000u && (uintptr_t)puVar5 < 0x900000u) {
         uVar2 = *puVar5;
         *puVar5 = puStack_14;
         *puStack_14 = uVar2;
+        }
       }
     }
     puStack_14 = puStack_14 + 0xd;
